main/options.c

```diff
@@ -1439,25 +1441,36 @@ static void processHelpOption (
 #ifdef HAVE_JANSSON
 static void processInteractiveOption (
 		const char *const option CTAGS_ATTR_UNUSED,
-		const char *const parameter CTAGS_ATTR_UNUSED)
+		const char *const parameter)
 {
+	struct interactiveModeArgs args;
+
 	Option.interactive = true;
+
+	if (parameter && (strcmp (parameter, "secure") == 0))
+	{
+#ifdef HAVE_SECCOMP
+		args.secure = true;
+#else
+		error (FATAL, "secure submode is not supported on this platform");
+#endif
+	}
+	else if (parameter && (strcmp (parameter, "nosecure") == 0))
+		args.secure = false;
+	else if ((!parameter) || *parameter == '\0')
+		args.secure = false;
+	else
+		error (FATAL, "Unknown option argument \"%s\" for --%s option",
+			   parameter, option);
+
```

Comment:
I strongly recommend sticking to my original, it is simpler. Less code = less room for errors.